resources/clj-kondo.exports/tensegritics/clojuredart/hooks/flutter.clj

```diff
@@ -0,0 +1,54 @@
+(ns hooks.flutter
+  (:require [clj-kondo.hooks-api :as api]
+            [clojure.set :as set]
+            [clojure.string :as str]))
+
+(defn widget 
+  "Hook for cljd.flutter.alpha/widget macro"
+  [{:keys [:node]}]
+    (let [args (rest (:children node))
+          opts-nodes (take-while (comp api/keyword-node? first) (partition 2 args))
+          pairs->value (fn [k] (first (filter #(-> % first (api/sexpr) (= k)) opts-nodes)))
+          [_ state] (pairs->value :state)
+          [_ context] (pairs->value :context)
+          [_ watch] (pairs->value :watch)
+          [_ key] (pairs->value :key)
+
+          bindings-count (* 2 (count opts-nodes))
+          body (drop bindings-count args)
+          error (fn [msg] (throw (ex-info msg {})))
+          unknown-keys (set/difference 
+                         (into #{} (map (comp api/sexpr first) opts-nodes))
+                         #{:state :key :watch :context})]
+
+      (cond
+        (and (seq (filter api/keyword-node? args))   ; has top keywords
+             (not (api/keyword-node? (first args)))) ; but first form is not a keyword
+        (error (str "Unknown first form " (api/sexpr (first args))))
+
+        (and state watch)
+        (error "State and watch keys are mutually exclusive")
+
+        (and state (not (api/vector-node? state)))
+        (error "State should be a vector [name initial-value]")
+
+        (and state (not= (count (:children state)) 2))
+        (error "State should contain only one name-value pair")
+
+        (seq unknown-keys)
+        (error (str "Unknown keys " (str/join " " unknown-keys)))
```

Comment:
`"Unsupported option keys: "`

Reply:
It would look like
<img width="422" alt="image" src="https://user-images.githubusercontent.com/14236531/164402931-f00d0df3-9816-4be8-aa25-de59e3735a4b.png">

So may be remove ':' after keys? Or its fine?